src/org/stringtemplate/v4/NumberRenderer.java

```diff
@@ -46,6 +46,8 @@ public String toString(Object o, String formatString, Locale locale) {
         if ( formatString==null ) return o.toString();
         Formatter f = new Formatter(locale);
         f.format(formatString, o);
-        return f.toString();
+        String str = f.toString();
```

Comment:
:bulb: This code should be in a try/finally for exception-safe closing of the `Formatter` instance.